integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -417,59 +396,30 @@ class CarbonFileMetastore extends CarbonMetaStore {
     thriftWriter.open(FileWriteOperation.OVERWRITE)
     thriftWriter.write(thriftTableInfo)
     thriftWriter.close()
-    updateSchemasUpdatedTime(touchSchemaFileSystemTime())
+    updateSchemasUpdatedTime(thriftTableInfo.getFact_table.getTable_id, thriftTableInfo
+      .getFact_table.getSchema_evolution.schema_evolution_history.get(0).time_stamp)
```

Comment:
Are you sure the evolution entry time is same as schema file update time?

Reply:
We dont actually need the file timestamp...schema evolution entry can also do the job because evertime the schema file is written a new entry for evolution is added to the schema.

Reply:
yes, it adds but it is not same as updated timestamp right? Since you are doing not equals check instead of greater than check it always refreshes